pkg/log/log.go

```diff
@@ -69,27 +67,19 @@ func (l *logger) Print(msg string) {
 			logFields[cf.Key()] = cf.Value()
 		}
 	}
-
-	l.entry = l.entry.WithFields(logFields)
-
-	switch l.level {
-	case InfoLevel:
-		l.entry.Info(msg)
-	case ErrorLevel:
-		l.entry.Error(msg)
-	case DebugLevel:
-		l.entry.Debug(msg)
+	entry := log.WithFields(logFields)
+	if l.err != nil {
+		entry = entry.WithError(l.err)
 	}
+	entry.Log(logrus.Level(l.level), msg)
 }
 
 func (l *logger) WithContext(ctx context.Context) Logger {
 	l.ctx = ctx
-	l.entry = l.entry.WithContext(ctx)
```

Comment:
@julio-lopez What do you think about setting the logrus.entry.Context using: `entry =entry.WithContext(l.ctx)` in the func `(l *logger) Print(msg string){}` ? We do extract the context fields though.

Reply:
The issues is that it would require keeping the entry around, instead of the context.
We may want to change the approach in the future and set the context in the entry, as you suggest, and then get the fields from the context at a much later stage, in the "hooks" or the "formatters". 
There is no need to set the context in the entry and keep it around, given the current approach where the fields are extracted and added to the entry before calling `Log(level, ...)`, as you point out. The fields need to be explicitly extracted, since `logrus` does not do anything special with the context itself. In particular, there is no context specific information in the log messages sent by default to stderr. This is because there is no general way to discover what values have been set in a context.